src/main/java/us/kilroyrobotics/subsystems/Tower.java

```diff
@@ -0,0 +1,288 @@
+        setState(TowerState.INIT);
+    }
+
+    public void runStateMachine() {
+        if (isTriggered(TowerEvent.HOME_TOWER)) setState(TowerState.INIT);
+        if (isTriggered(TowerEvent.SCORE_BYPASS)) {
+            coralIntakeMotor.setSpeed(CoralMechanismConstants.kWheelSpeedOuttaking);
+
+            setState(TowerState.SCORING);
+        }
+        switch (currentState) {
+            case INIT:
+                wrist.setAngle(CoralMechanismConstants.kIntakingAngle);
+                coralIntakeMotor.setSpeed(0.0);
+
+                setState(TowerState.HOMING_WRIST);
+                break;
+            case HOMING_WRIST:
+                if (wrist.inPosition()) setState(TowerState.HOMING_ELEVATOR);
+                break;
+            case HOMING_ELEVATOR:
+                if (elevator.inPosition()) {
+                    elevator.setPosition(ElevatorConstants.kZeroed);
+
+                    setState(TowerState.HOME);
+                }
+                break;
+            case HOME:
+                if (isTriggered(TowerEvent.INTAKE_CORAL)) {
+                    wrist.setAngle(CoralMechanismConstants.kIntakingAngle);
+
+                    setState(TowerState.TILTING_TO_INTAKE);
+                } else if (coralIntakeMotor.isCoralDetected()) setState(TowerState.GOT_CORAL);
+                else currentLevel = 0;
+                break;
+            case TILTING_TO_INTAKE:
+                if (wrist.inPosition()) {
+                    elevator.setPosition(ElevatorConstants.kCoralStationHeight);
+
+                    setState(TowerState.RAISING_TO_INTAKE);
+                }
+                break;
+            case RAISING_TO_INTAKE:
+                if (elevator.inPosition()) {
+                    coralIntakeMotor.setSpeed(CoralMechanismConstants.kWheelSpeedIntaking);
+                    leds.setMode(LEDMode.WaitingForCoral);
+
+                    setState(TowerState.INTAKING);
+                }
+                break;
+            case INTAKING:
+                if (coralIntakeMotor.isCoralDetected() || isTriggered(TowerEvent.INTAKE_BYPASS)) {
+                    // elevator.setPosition(ElevatorConstants.kZeroed);
+                    coralIntakeMotor.setSpeed(CoralMechanismConstants.kWheelSpeedHolding);
+                    leds.setMode(LEDMode.CoralDetected);
+
+                    setState(TowerState.GOT_CORAL);
+                }
+                break;
+            case GOT_CORAL:
+                if (isTriggered(TowerEvent.GOTO_L1)) {
+                    elevator.setPosition(ElevatorConstants.kL1Height);
+                    coralIntakeMotor.setSpeed(CoralMechanismConstants.kWheelSpeedHolding);
+
+                    currentLevel = 1;
+                    setState(TowerState.RAISING_TO_L1);
+                } else if (isTriggered(TowerEvent.GOTO_L2)) {
+                    elevator.setPosition(ElevatorConstants.kL2Height);
+                    wrist.setAngle(CoralMechanismConstants.kScoringL2);
```

Comment:
You're doubling up on this, this is done in RAISING_TO_L2

Reply:
This was modified because L2 needs to do both at the same time, not sequential. It fails to score otherwise because the scoral mechanism hits the L3 peg.

Reply:
Remove it from RAISING_TO_L2 then